src/basic/env-util.c

```diff
@@ -33,7 +33,7 @@ static bool env_name_is_valid_n(const char *e, size_t n) {
         if (n == SIZE_MAX)
                 n = strlen_ptr(e);
 
-        if (n <= 0)
+        if (n == 0)
```

Comment:
IIRC, @poettering prefers `<=` even for unsigned. I do not have any preference on it. So, feel free to keep it.

Reply:
FWIW, I find `<=` for unsigned types *very* confusing. I always do a double take and recheck that the type is indeed unsigned. Variables don't change type unexpectedly, so this kind of "safe coding" just in case somebody comes and changes the type of the variable we define ourselves seems completely unneeded.

The only place where this would make sense is for externally defined variables in libc and elsewhere where there is some confusion as to the exact type, or the documented type is iffy.